### ISSUE TITLE
Control-flow constructs inside C-style varargs crash the frontend

### ISSUE THREAD

**Issue by densh** [scala-native/scala-native#153]
``` scala
import scalanative.native._
import scalanative.libc.stdlib._

object Test {
  def main(args: Array[String]): Unit = {
    fprintf(stdout, c"%d", if ("".isEmpty) 0 else 1)
  }
}
```

Crashes with:

```
scala.MatchError: (Focus(List(),Local(src,2),List(Local(Local(src,0),Module(Top(Test$))), Local(Local(src,1),Class(Top(s
cala.scalanative.runtime.ObjectArray)))),List(Inst(Local(src,3),Module(Top(scala.scalanative.libc.stdlib$))), Inst(Local
(src,4),Load(Ptr,Global(Member(Top(scala.scalanative.libc.stdlib$),extern.__stdoutp),Ptr)))),Const(Chars(%d)),false),if
("".isEmpty())
  native.this.Vararg.apply(scala.Int.box(0), (ClassTag.Int(): scala.reflect.ClassTag))
else
  native.this.Vararg.apply(scala.Int.box(1), (ClassTag.Int(): scala.reflect.ClassTag))) (of class scala.Tuple2)
        at scala.scalanative.nscplugin.NirCodeGen$SaltyCodePhase$$anonfun$31.apply(NirCodeGen.scala:1533)
        at scala.scalanative.nscplugin.NirCodeGen$SaltyCodePhase$$anonfun$31.apply(NirCodeGen.scala:1533)
        at scala.collection.LinearSeqOptimized$class.foldLeft(LinearSeqOptimized.scala:124)
        at scala.collection.immutable.List.foldLeft(List.scala:84)
        at scala.scalanative.nscplugin.NirCodeGen$SaltyCodePhase.genExpandRepeatedArg(NirCodeGen.scala:1533)
        at scala.scalanative.nscplugin.NirCodeGen$SaltyCodePhase$$anonfun$genMethodArgs$2.apply(NirCodeGen.scala:1513)
        at scala.scalanative.nscplugin.NirCodeGen$SaltyCodePhase$$anonfun$genMethodArgs$2.apply(NirCodeGen.scala:1510)
        at scala.collection.TraversableLike$WithFilter$$anonfun$foreach$1.apply(TraversableLike.scala:733)
        at scala.collection.immutable.List.foreach(List.scala:381)
        at scala.collection.TraversableLike$WithFilter.foreach(TraversableLike.scala:732)
        at scala.scalanative.nscplugin.NirCodeGen$SaltyCodePhase.genMethodArgs(NirCodeGen.scala:1510)
        at scala.scalanative.nscplugin.NirCodeGen$SaltyCodePhase.genMethodCall(NirCodeGen.scala:1474)
        at scala.scalanative.nscplugin.NirCodeGen$SaltyCodePhase.genMethodCall(NirCodeGen.scala:1462)
        at scala.scalanative.nscplugin.NirCodeGen$SaltyCodePhase.genApply(NirCodeGen.scala:821)
        at scala.scalanative.nscplugin.NirCodeGen$SaltyCodePhase$$anonfun$genExpr$1.apply(NirCodeGen.scala:449)
        at scala.scalanative.nscplugin.NirCodeGen$SaltyCodePhase$$anonfun$genExpr$1.apply(NirCodeGen.scala:407)
        at scala.scalanative.nscplugin.NirCodeGen$SaltyCodePhase.notMergeableGuard(NirCodeGen.scala:368)
        at scala.scalanative.nscplugin.NirCodeGen$SaltyCodePhase.genExpr(NirCodeGen.scala:406)
        at scala.scalanative.nscplugin.NirCodeGen$SaltyCodePhase$$anonfun$17.apply(NirCodeGen.scala:675)
        at scala.scalanative.nscplugin.NirCodeGen$SaltyCodePhase$$anonfun$17.apply(NirCodeGen.scala:675)
        at scala.scalanative.nir.Focus$$anonfun$sequenced$1.apply(Focus.scala:152)
        at scala.scalanative.nir.Focus$$anonfun$sequenced$1.apply(Focus.scala:151)
        at scala.collection.LinearSeqOptimized$class.foldLeft(LinearSeqOptimized.scala:124)
        at scala.collection.immutable.List.foldLeft(List.scala:84)
        ...
```

**Comment by densh** [comment]
**Workaround:** move the control flow expression out into a local val.

**Comment by densh** [comment]
Fixed in #1426
